### DISCUSSION TITLE
Keyball61 衝撃後の右手側縦2列のキー無効について

### DISCUSSION THREAD

**Discussion by ShinjiKS0209** [Yowkees/keyball#368] [Q&A]
Yowkees様

ShinjiKS0209と申します。
掲題の件について、ご助言いただけないでしょうか。
ビルド完了して２週間ほど使用してきたものの、先日右手側の淵に右手をひっかけて数センチ浮かして落とすような衝撃を加えてしまいました。
その後、画像赤枠のキーが効かなくなってしまい、同様の症状かと思われるポストを参考に、画像青枠の一部のダイオードをはんだ付けしなおしたものの症状は変わりませんでした。
なお、各キーのキーソケットについては押した状態でショートすることを確認しており、キーソケットやキースイッチに問題はないかと認識しております。
構成は以下です。
購入モデル：keyball 61 rev2
購入先：遊舎工房
左手：トラックボール
右手：標準

ダイオードのはんだ付けが不適切か、あるいは他の原因が考えられるか、ご助言いただけると助かります。
![PXL_20230930_014117298異常個所枠追加](https://github.com/Yowkees/keyball/assets/108440600/dd625685-029a-4b5d-b7a4-2b1edaed234a)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

衝撃でキーが効かなくなったということで、もともと半田付け不良でギリギリ導通していたものが離れてしまった感じだと思います。

ダイオードですが、緑丸をつけたものがFキーの縦１列に関連していますので、半田付けをご確認ください。

ここまで書いて気付きましたが、不具合のある２列はどちらもCOL1という信号線で読み取るキーですね。
ということで、下記画像の緑点をつけた端子がうまく読み取れていないようです。緑点の端子の半田付けをご確認ください。

また、ProMicro側の半田付けが原因かも知れませんので、左右のProMicroを入れ替えて、どのような症状となるかご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

![KB61L_shock_fix](https://github.com/Yowkees/keyball/assets/84209772/e5e5f5c5-39db-4ee7-9ed6-e5bd35cd9242)

**Reply by ShinjiKS0209** [reply]
ご回答ありがとうございます。
ProMicroを左右で付け替えたら全く逆の事象が発生しました。
症状が出ていた右手側と対称に位置する左手側のキーが効かなくなり、右手側はすべて機能しました。
ということで、ProMicroのはんだ？に原因があると切り分けできましたかと思います。
ご指摘いただいてるのはキーボード基盤の右側上から6番目のCol1課と思いますが、ProMicro側の対象箇所を添付画像にて赤丸で囲んでいます。左右両方のProMicroを外して、それぞれ元々つけていたキーボード基盤と並べて撮影しております。
パッと見では、はんだ付けが外れてる感じはしませんが、はんだの量が少ないなどあればご指摘いただけると助かります。
![PXL_20231007_020952769原因箇所枠追加](https://github.com/Yowkees/keyball/assets/108440600/32622e0f-4c9d-48e1-aaa5-dfb29023c61f)

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

半田付けには特に問題はなさそうです。
テスター等で隣とショートしていないかご確認いただき、問題なさそうでしたら、ProMicroの初期不良の可能性もあります。
Keyballシリーズご購入者様限定で、12Pピンヘッダー単体での販売も可能ですので、ご検討ください。

以上、お手数ですがご対応お願いいたします。
